addons/mail/models/discuss/discuss_channel.py

```diff
@@ -904,28 +935,7 @@ def _get_channels_as_member(self):
         channels += self.env["discuss.channel"].search(pinned_member_domain)
         return channels
 
-    def _channel_basic_info(self):
-        self.ensure_one()
-        data = self._read_format(
-            [
-                "allow_public_upload",
-                "avatar_cache_key",
-                "channel_type",
-                "create_uid",
-                "default_display_mode",
-                "description",
-                "last_interest_dt",
-                "member_count",
-                "name",
-                "uuid",
-            ],
-            load=False,
-        )[0]
-        data["authorizedGroupFullName"] = self.group_public_id.full_name
-        data["group_based_subscription"] = bool(self.group_ids)
-        return data
-
-    def _to_store_defaults(self):
+    def _to_store_defaults(self, for_current_user=True):
```

Comment:
Note: added `for_current_user` flag, when broadcasting (without using `with_user`) it should always be set to False to avoid leaking data.

This was due to the removing of `_channel_basic_info`, store is now used in more contexts where data for current user should not be sent.